### PR TITLE
[C] Challenge 14 (Reviewed)

### PR DESCRIPTION
**Pull request by karanchawla**

### CODE REVIEW

File: challenge_14/c/karanchawla/challenge_14.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+{
+	char c;
+	struct node* next;
+}Node;
+
+
+//utility function to create a new node
+Node* newNode(char c)
+{
+	Node* NewNode = (Node*)malloc(sizeof(Node));
```
Comment (erocs) [inline challenge_14/c/karanchawla/challenge_14.c]: Try to maintain consistency in your naming convention. Here you use camel case for a variable but below you use a more traditional lowercase with underscores.

File: challenge_14/c/karanchawla/challenge_14.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	return NewNode;
+}
+
+//utility function to add a new node to the linked list
+void push(Node** head_ref, char c)
+{
+	Node* new_node = newNode(c);
+
+	new_node->next = (*head_ref);
```
Comment (erocs) [inline challenge_14/c/karanchawla/challenge_14.c]: I'm curious, why are you wrapping the de-references in parenthesis? It isn't needed and looks a little odd.

File: challenge_14/c/karanchawla/challenge_14.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+//utility function to print linked list
+void printLinkedList(Node* head)
+{
+	while(head!=NULL)
+	{
+		printf("%c->", head->c);
+		head = head->next;
+	}
+	printf("NULL");
```
Comment (erocs) [inline challenge_14/c/karanchawla/challenge_14.c]: You likely want to also print a newline at the end, "NULL\n".

File: challenge_14/c/karanchawla/challenge_14.c (hunk 1 above)
Comment (karanchawla) [inline challenge_14/c/karanchawla/challenge_14.c]: Yeah, I don't know why I did this - usually, I'm pretty careful of such errors.

File: challenge_14/c/karanchawla/challenge_14.c (hunk 3 above)
Comment (karanchawla) [inline challenge_14/c/karanchawla/challenge_14.c]: Made the changes. And yes shouldn't wrap my de-refs in parenthesis. Thanks for the review :)

**Pull request merged**